### PR TITLE
Doc - multi gateway deployments

### PR DESCRIPTION
**Pull request by bcollard**
# Description

Guide for multi-gateway deployments.
Explains the purpose of Proxies and Gateways.
Comes with a full example.

# Context

Users ran into this bug doing ... \ Users needed this feature to ...

# Checklist:

- [ ] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [ ] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [ ] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [ ] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: docs/content/installation/advanced_configuration/multi-gw-deployment.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+In the Helm values below, we will define two Envoy proxies, named `publicGw` and `corpGw`. One could be internet facing, whereas the latter could be for intranet usage.
+
+By default, for each Proxy, Helm will create two `Gateways`: one for HTTP and another one for HTTPS. \
+In our example, we will disable the HTTP `Gateway` for the internet-facing Proxy, and also disable the HTTPS Gateway for the traffic coming from the Intranet.
+
+Overview:
+
+![Full example overview]({{< versioned_link_path fromRoot="/img/gw-proxies-full-example.png" >}})
+
+If you want additional `Gateways` for a given Proxy, consider crafting `Gateway` _Custom Resources_ youself, similarly to what you can do with `VirtualServices`.
```
Comment (artberger) [inline docs/content/installation/advanced_configuration/multi-gw-deployment.md:69]: Is there a link we can provide for how they craft a Gateway custom resource? Maybe the API reference?
```suggestion
If you want additional `Gateways` for a single proxy, create your own `Gateway` _Custom Resources_, similar to what you can do with `VirtualServices`. For more information, see the [`Gateway` API reference documentation]({{< versioned_link_path fromRoot="/reference/api/github.com/solo-io/gloo/projects/gateway/api/v1/gateway.proto.sk/" >}})
```

File: docs/content/installation/advanced_configuration/multi-gw-deployment.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+As you will see in the example below, you can declare as many Envoy proxies as you want under the Helm's `gloo.gatewayProxies` property.
```
Comment (artberger) [inline docs/content/installation/advanced_configuration/multi-gw-deployment.md:71]:
```suggestion
As shown in the following example, you can declare as many Envoy proxies as you want under the `gloo.gatewayProxies` property in the Helm configuration file.
```

**Review by artberger — COMMENTED** [review]
Thanks so much for adding this multi-gateway example to the docs! I added some suggestions from a docs-perspective for grammar, style, clarity, and consistency with the terms and formatting.

File: docs/content/installation/advanced_configuration/multi-gw-deployment.md (hunk 1 above)
Comment (bcollard) [reply docs/content/installation/advanced_configuration/multi-gw-deployment.md:66]: Perfect! 👌

File: docs/content/installation/advanced_configuration/multi-gw-deployment.md [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+You can use the following Helm configuration file to create multiple proxies.
+* `publicGw`: An internet-facing proxy, with the default HTTP `Gateway` disabled so that only secure HTTPS traffic is allowed from the public network.
+* `corpGw`: A proxy for the company intranet, with the default HTTPS `Gateway` disabled so that traffic does not have to be encrypted because the network is private.
+
+Overview diagram:
+
+![Full example overview]({{< versioned_link_path fromRoot="/img/gw-proxies-full-example.png" >}})
+
+If you want additional `Gateways` for a single proxy, create your own `Gateway` _Custom Resources_, similar to what you can do with `VirtualServices`. For more information, see the [`Gateway` API reference documentation]({{< versioned_link_path fromRoot="/reference/api/github.com/solo-io/gloo/projects/gateway/api/v1/gateway.proto.sk/" >}})
```
Comment (artberger) [inline docs/content/installation/advanced_configuration/multi-gw-deployment.md:65]: I forgot the period.
```suggestion
If you want additional `Gateways` for a single proxy, create your own `Gateway` _Custom Resources_, similar to what you can do with `VirtualServices`. For more information, see the [`Gateway` API reference documentation]({{< versioned_link_path fromRoot="/reference/api/github.com/solo-io/gloo/projects/gateway/api/v1/gateway.proto.sk/" >}}).
```

**Review by artberger — APPROVED** [review]

**Pull request merged**